docs/experimental/howtos/iterate_prompt.md

```diff
@@ -0,0 +1,329 @@
+# How to Evaluate Your Prompt and Improve It
+
+In this guide, you'll learn how to evaluate and iteratively improve a prompt using Ragas.
+
+## What you'll accomplish
+- Evaluate your prompt on a dataset
+- Analyze errors
+- Improve prompt based on the experiment
+- Evaluate the new prompt
+- Choose the final prompt based on results
+
+!!! note "Full code"
+    - The dataset and scripts live under `examples/iterate_prompt/` in the repo
+    - Full code is available on [GitHub](https://github.com/explodinggradients/ragas/tree/main/examples/iterate_prompt)
+
+## Task definition
+In this case, we are considering a customer support ticket classification task.
+
+- Labels (multi-label): `Billing`, `Account`, `ProductIssue`, `HowTo`, `Feature`, `RefundCancel`
+- Priority (exactly one): `P0`, `P1`, or `P2`
+
+
+## Dataset 
+
+We've created a synthetic dataset for our use case. Each row has `id, text, labels, priority`. Example rows from the dataset:
+
+```csv
+id,text,labels,priority
+1,"…duplicate charge…","Billing;RefundCancel",P1
+2,"…SSO bounces to /login…","Account;ProductIssue",P0
+```
+
+To customize the dataset for your use case, create a `datasets/` directory and add your own CSV file. You can also connect to different backends. Refer to [Datasets - Core Concepts](../core_concepts/datasets.md) for more information.
+
+It is better to sample real data from your application to create the dataset. If that is not available, you can generate synthetic data using an LLM. We recommend using a reasoning model like gpt-5 high-reasoning which can generate more accurate and complex data. Always make sure to manually review and verify the data you use. 
+
+## Evaluate your prompt on a dataset
+
+### Prompt runner
+
+First, we'll run the prompt on one case to test if everything works. 
+
+```bash
+cd examples/iterate_prompt
+export OPENAI_API_KEY=your_openai_api_key
+uv run run_prompt.py
+```
+
+This will run the prompt on sample case and print the results.
+
+??? example "Sample output"
+    ```
+    $ uv run run_prompt.py                      
+
+    Test ticket:
+    "SSO via Okta succeeds then bounces me back to /login with no session. Colleagues can sign in. I tried clearing cookies; same result. Error in devtools: state mismatch. I'm blocked from our boards."
+
+    Response:
+    {"labels":["Account","ProductIssue"], "priority":"P0"}
+    ```
+
+
+### Metrics for scoring
+
+It is generally better to use a simpler metric instead of a complex one. You should use a metric relevant to your use case. More information on metrics can be found in [Metrics - Core Concepts](../core_concepts/metrics.md). Here we are looking at labels_exact_match and priority_accuracy. We want to keep it separate so that we can analyze errors for each metric separately and improve them.
```

Comment:
Add two bullet points briefly explaining each metric (why it's important for this task, and what it does)

Reply:
Also you can use `priority_accuracy` format while adding such items. For example
- `priority_accuracy`:....
- `labels_exact_match`:...